File: code/community/1304596979887640578/503-504-errors-deepgram-api.py
import os
import requests

# Retrieve environment variables
DEEPGRAM_API_KEY = os.getenv('DEEPGRAM_API_KEY')
PROJECT_ID = os.getenv('PROJECT_ID')

# Endpoint for Deepgram transcription
url = 'https://api.deepgram.com/v1/listen'

def transcribe_audio(audio_url):
    headers = {
        'Authorization': f'Token {DEEPGRAM_API_KEY}',
        'Content-Type': 'application/json'
    }
    data = {
        'url': audio_url,
        'project': PROJECT_ID
    }
    response = requests.post(url, headers=headers, json=data)
    if response.status_code == 200:
        print('Transcription:', response.json())
    else:
        print('Error:', response.status_code, response.text)

# Example audio URL
audio_url = 'https://example.com/audiofile.mp3'

transcribe_audio(audio_url)

File: code/community/1304596979887640578/503-504-errors-deepgram-api.js
require('dotenv').config();
const fetch = require('node-fetch');

const DEEPGRAM_API_KEY = process.env.DEEPGRAM_API_KEY;
const PROJECT_ID = process.env.PROJECT_ID;

const url = 'https://api.deepgram.com/v1/listen';

async function transcribeAudio(audioUrl) {
    const response = await fetch(url, {
        method: 'POST',
        headers: {
            'Authorization': `Token ${DEEPGRAM_API_KEY}`,
            'Content-Type': 'application/json'
        },
        body: JSON.stringify({
            url: audioUrl,
            project: PROJECT_ID
        })
    });

    if (response.ok) {
        const data = await response.json();
        console.log('Transcription:', data);
    } else {
        console.error('Error:', response.status, await response.text());
    }
}

// Example audio URL
const audioUrl = 'https://example.com/audiofile.mp3';

transcribeAudio(audioUrl);